### PR TITLE
ci: fix macos builds

### PR DESCRIPTION
**Pull request by gnufede**
CI: Fix MacOS builds
 - Pin MacOS runner to 12 instead of latest
 - Pin cmake as 3.28 hangs

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by avara1986 — APPROVED** [review]

**Review by christophe-papazian — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools_scm[toml]>=4", "cython", "cmake>=3.24.2; python_version>='3.7'"]
+requires = ["setuptools_scm[toml]>=4", "cython", "cmake<3.28; python_version>='3.7'"]
```
Comment (mabdinur) [inline pyproject.toml:2]: Should we keep the minimum version?
```suggestion
requires = ["setuptools_scm[toml]>=4", "cython", "cmake>=3.24.2,<3.28; python_version>='3.7'"]
```

File: setup.py [hunk 2]
```diff
@@ -618,7 +618,7 @@ def get_ddup_ext():
         "Programming Language :: Python :: 3.11",
         "Programming Language :: Python :: 3.12",
     ],
-    setup_requires=["setuptools_scm[toml]>=4", "cython", "cmake>=3.24.2"],
+    setup_requires=["setuptools_scm[toml]>=4", "cython", "cmake<3.28"],
```
Comment (mabdinur) [inline setup.py:621]: same here

**Review by mabdinur — APPROVED** [review]
Left some non-blocking comments

File: setup.py (hunk 2 above)
Comment (gnufede) [inline setup.py:621]:
```suggestion
    setup_requires=["setuptools_scm[toml]>=4", "cython", "cmake>=3.24.2,<3.28"],
```

**Review by emmettbutler — APPROVED** [review]

**Review by juanjux — APPROVED** [review]

**Pull request merged**